File: scripts/crons/cron4.js
var NextPayments = require('payparrot_models/objects/next_payments.js'),
	Suscriptions = require('payparrot_models/objects/suscriptions'),
	db = require('payparrot_models/libs/mongodb').connect(),
	async = require('async'),
	queue = require('payparrot_models/libs/queue'),
	db = require('payparrot_models/libs/mongodb').connect({});

var queue_next_payment = function(payment, callback){
	queue.createMessage('payments', {
			MessageBody: JSON.stringify({'suscription_id': payment.suscription_id, 'account_id': payment.account_id, 'parrot_id': payment.parrot_id})
		}, function(err){
			if(!err){
				NextPayments.remove({_id: payment._id}, function(err){
					callback();
				});
			}else{
				callback();
			}
		}
	);
};

var get_suscription = function(payment, callback){
	Suscriptions.findOne({'account_id': payment.account_id, 'parrot_id': payment.parrot_id}, {}, function (err, suscription){
		try{
			if(!err && suscription){
				payment.suscription_id = suscription._id;
				queue_next_payment(payment, callback);
			}else{
				callback();
			}
		}catch(e){
			callback();
		}
	});
};

NextPayments
	.find({},{'parrot_id':1,'account_id':1,'_id':1})
	.sort('_id', 'descending')
	.where('action_date')
	.lte(new Date())
	.run(function (err, next_payments){
			console.log(next_payments);
			console.log(new Date());
			async.forEach(next_payments, function(item, callback){get_suscription(item, callback)}, function(err){
				db.connection.close();
			});
	});

File: scripts/payparrot_scripts/crons/cron4.py
# -*- coding: utf-8 -*-
from datetime import datetime

from payparrot_dal import Subscriptions, NextPayments
from payparrot_dal.mongodb import connect
from payparrot_dal.queue import Queue
from payparrot_scripts.crons.utils import log

def main():
    connection = None
    log('cron4', 'Starting')
    try:
        connection, db = connect()
        next_payments = NextPayments.find(
            db, {'action_date': {'$lte': datetime.now()}}, {'parrot_id':1,'account_id':1,'_id':1}
        ).sort([('_id', -1)])
        
        for next_payment in next_payments:
            subscription = Subscriptions.findOne(db, {'account_id': next_payment.get('account_id'), 'parrot_id': next_payment.get('parrot_id')})
            if subscription and subscription.active:
                created_message = Queue.insert(
                    'payments', 
                    {
                        'subscription_id': str(subscription.id),
                        'account_id': str(subscription.account_id),
                        'parrot_id': str(subscription.parrot_id)
                    }
                )
                if created_message:
                    db.next_payments.remove({'_id': next_payment.get('_id')})
            else:
                db.next_payments.remove({'_id': next_payment.get('_id')})
    finally:
        if connection:
            connection.close()
        log('cron4', 'Finishing')

if __name__ == '__main__':
    main()            